extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -268,6 +303,62 @@ export abstract class AzureAuth implements vscode.Disposable {
 		return this.handleInteractionRequired(tenant, resource);
 	}
 
+	/**
+	 * Gets the access token for the correct account and scope from the token cache, if the correct token doesn't exist in the token cache
+	 * (i.e. expired token, wrong scope, etc.), sends a request for a new token using the refresh token
+	 * @param accountId
+	 * @param azureResource
+	 * @returns The authentication result, including the access token
+	 */
+	// TODO: Need to add resource we are getting the token for
+	// Can replace most refreshToken calls with getToken, it does the same thing
+	// TODO: separate getToken for MSAL
+	public async getTokenMsal(accountId: string, azureResource: azdata.AzureResource): Promise<AuthenticationResult> {
+		const cache = this.clientApplication.getTokenCache();
+		if (!cache) {
+			console.log('Error: Could not fetch token cache.');
+		}
+		const resource = this.resources.find(s => s.azureResourceId === azureResource);
+		if (!resource) {
+			console.log('Error: Could not fetch the azure resource.');
```

Comment:
Make sure any log messages you have include any information that will be useful in debugging, for example here it'd be useful to include not only the item you're trying to find but the items in the list you're searching to help with debugging.